src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorCodeGenerationOptionsBuilder.cs

```diff
@@ -58,51 +80,110 @@ public abstract class RazorCodeGenerationOptionsBuilder
+    public bool SuppressNullabilityEnforcement
+    {
+        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.SuppressNullabilityEnforcement);
+        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.SuppressNullabilityEnforcement, value);
+    }
 
     /// <summary>
     /// Gets or sets a value that determines if the components code writer may omit values for minimized attributes.
     /// </summary>
-    public virtual bool OmitMinimizedComponentAttributeValues { get; set; }
+    public bool OmitMinimizedComponentAttributeValues
+    {
+        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.OmitMinimizedComponentAttributeValues);
+        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.OmitMinimizedComponentAttributeValues, value);
+    }
 
     /// <summary>
     /// Gets or sets a value that determines if localized component names are to be supported.
     /// </summary>
-    public virtual bool SupportLocalizedComponentNames { get; set; }
+    public bool SupportLocalizedComponentNames
+    {
+        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.SupportLocalizedComponentNames);
+        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.SupportLocalizedComponentNames, value);
+    }
 
     /// <summary>
     /// Gets or sets a value that determines if enhanced line pragmas are to be utilized.
     /// </summary>
-    public virtual bool UseEnhancedLinePragma { get; set; }
+    public bool UseEnhancedLinePragma
+    {
+        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.UseEnhancedLinePragma);
+        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.UseEnhancedLinePragma, value);
+    }
 
     /// <summary>
     /// Gets or sets a value that determines if unique ids are suppressed for testing.
     /// </summary>
-    internal string SuppressUniqueIds { get; set; }
+    public string? SuppressUniqueIds { get; set; }
 
     /// <summary>
     /// Determines whether RenderTreeBuilder.AddComponentParameter should not be used.
     /// </summary>
-    internal bool SuppressAddComponentParameter { get; set; }
+    public bool SuppressAddComponentParameter
+    {
+        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.SuppressAddComponentParameter);
+        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.SuppressAddComponentParameter, value);
+    }
 
     /// <summary>
     /// Determines if the file paths emitted as part of line pragmas should be mapped back to a valid path on windows.
     /// </summary>
-    internal bool RemapLinePragmaPathsOnWindows { get; set; }
+    public bool RemapLinePragmaPathsOnWindows
+    {
+        get => _flags.IsFlagSet(RazorCodeGenerationOptionsFlags.RemapLinePragmaPathsOnWindows);
+        set => _flags.UpdateFlag(RazorCodeGenerationOptionsFlags.RemapLinePragmaPathsOnWindows, value);
+    }
+
+    public RazorCodeGenerationOptionsBuilder(RazorConfiguration configuration)
```

Comment:
to be consistent might wanna add triple slash docs for these ones too

Reply:
I don't think XML doc comments would add much value here. These are not _actually_ public APIs. We no longer support a public API on the Razor compiler or ship it on NuGet, so the XML doc comments are essentially developer notes.